pkg/odo/cli/component/ui/ui.go

```diff
@@ -0,0 +1,228 @@
+	var path string
+	prompt := &survey.Input{
+		Message: fmt.Sprintf("Where does the %s input for the component reside on the local file system", inputType),
+	}
+	if len(defaultPath) > 0 {
+		prompt.Default = defaultPath
+	}
+	err := survey.AskOne(prompt, &path, validation.PathValidator)
+	ui.HandleError(err)
+	return path
+}
+
+// we need this because the validator for the component name needs use info from the Context
+// so we effectively return a closure that references the context
+func createComponentNameValidator(context *genericclioptions.Context) survey.Validator {
+	return func(input interface{}) error {
+		if s, ok := input.(string); ok {
+			err := validation.ValidateName(s)
+			if err != nil {
+				return err
+			}
+
+			exists, err := component.Exists(context.Client, s, context.Application)
+			if err != nil {
+				glog.V(4).Info(err)
+				return fmt.Errorf("Unable to determine if component '%s' exists or not", s)
+			}
+			if exists {
+				return fmt.Errorf("Component with name '%s' already exists in application '%s'", s, context.Application)
+			}
+
+			return nil
+		}
+
+		return fmt.Errorf("can only validate strings, got %v", input)
+	}
+}
+
+// EnterComponentName allows the user to specify the component name in a prompt
+func EnterComponentName(defaultName string, context *genericclioptions.Context) string {
+	var path string
+	prompt := &survey.Input{
+		Message: "How would you wish to name the new component",
+		Default: defaultName,
+	}
+	err := survey.AskOne(prompt, &path, createComponentNameValidator(context))
+	ui.HandleError(err)
+	return path
+}
+
+// EnterGitInfo will display two prompts, one of the URL of the project and one of the ref
+func EnterGitInfo() (string, string) {
+	gitURL := enterGitInputTypePath()
+	gitRef := enterGitRef("master")
+
+	return gitURL, gitRef
+}
+
+func enterGitInputTypePath() string {
+	var path string
+	prompt := &survey.Input{
+		Message: "What is the URL of the git repository you would wish the new component to use",
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+func enterGitRef(defaultRef string) string {
+	var path string
+	prompt := &survey.Input{
+		Message: "What git ref (branch, tag, commit) would you wish to use",
+		Default: defaultRef,
+	}
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+	return path
+}
+
+// Proceed displays a given message and asks the user if they want to proceed
+func Proceed(message string) bool {
+	var response bool
+	prompt := &survey.Confirm{
+		Message: message,
+	}
+	err := survey.AskOne(prompt, &response, survey.Required)
+	ui.HandleError(err)
+
+	return response
+}
+
+// EnterPorts allows the user to specify the ports to be used in a prompt
+func EnterPorts() []string {
+	var portsStr string
+	prompt := &survey.Input{
+		Message: "Enter the ports you wish to set (for example: 8080,8100/tcp,9100/udp)",
```

Comment:
You should add here that you can just press enter / not input anything if you want to bypass this setup.

Reply:
Fixed